src/pspm_extract_segments.m

```diff
@@ -1,680 +1,738 @@
-                options = varargin{5};
-            else
-                options = struct();
-            end;
-            
-            % expand chan if necessary
-            if iscell(data_fn) && isnumeric(chan) ...
-                    && numel(chan) == 1
-                chan = repmat({chan}, size(data_fn));
-            end;
-            
-            % check if specific arguments are correct
-            if ~ischar(data_fn)
-                if (iscell(data_fn) && any(~cellfun(@ischar, data_fn))) ||~iscell(data_fn)
-                    warning('ID:invalid_input', 'data_fn has to be a string or a cell array of strings.'); return;
+    % pspm_extract_segments. Function in order to extract segments of a certain
+    % length after defined onsets and create summary statistics
+    % (mean, SD, SEM, % NaN) over these segments
+    %
+    % The function supports either manual setting of data files, channels,
+    % timing and timeunits or automatic extraction from a glm model file.
+    %
+    % The segments variable returned will be a cx1 cell where c corresponds to
+    % the number of conditions. Each element contains a struct with
+    % fields data, mean, std and sem (std of the mean).
+    % The field data is a nxo*s vector where n is number of data points and o*s
+    % corresponds to the onsets multiplied by the sessions.
+    %
+    %   FORMAT:
+    %       [sts, segments] = pspm_extract_segments('manual', data_fn, chan, timing, options)
+    %       [sts, segments] = pspm_extract_segments('manual', data_raw, sr, timing, options)
+    %       [sts, segments] = pspm_extract_segments('auto', glm, options)
+    %       [sts, segments] = pspm_extract_segments('auto', dcm, options)
+    %
+    %   ARGUMENTS:
+    %       mode:               Tells the function in which mode get the
+    %                           settings from. Either 'manual' or 'auto'.
+    %       glm:                Path to the glm file or a glm structure.
+    %       dcm:                Path to the dcm file or a dcm structure.
+    %       data_fn:            Path or cell of paths to data files from which
+    %                           the segments should be extracted. Each file
+    %                           will be treated as session. Onset values are
+    %                           averaged through conditions and sessions.
+    %       data_raw:           Numeric raw data or a cell array of numeric raw data.
+    %       chan:               Channel number or cell of channel numbers which
+    %                           defines which channel should be taken to
+    %                           extract the segments. Chan should correspond to
+    %                           data_fn and should have the same length. If
+    %                           data_fn is a cell and chan is a single number,
+    %                           the number will be taken for all files.
+    %       sr:                 Array of sampling rates (same dimension as data_raw)

```

Comment:
Writing "same dimension as `data_raw`" might be confusing, so we should note that this is the case only if `data_raw` is a cell array.

Reply:
Copy that